client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,383 @@ import {
+	}
+	return (
+		<button
+			className="web-payment-box__web-pay-button button checkout__pay-button-button button is-primary button-pay pay-button__button"
+			onClick={ onClick }
+		>
+			{ translate( 'Select a payment card' ) }
+		</button>
+	);
+}
 
-	/**
-	 * @param {string} fieldName The name of the country select field.
-	 * @param {string} selectedCountryCode The selected country.
-	 */
-	updateSelectedCountry = ( fieldName, selectedCountryCode ) => {
-		setTaxCountryCode( selectedCountryCode );
-
-		// Apple Pay needs the country where the payment will be processed
-		// (https://developer.apple.com/documentation/apple_pay_on_the_web/applepayrequest/2951833-countrycode),
-		// so call the Paygate configuration API endpoint to get that. This
-		// information is only needed when the user clicks the button to buy
-		// something and opens the Apple Pay payment sheet, so it is somewhat
-		// wasteful to call it here (every time a new country is selected);
-		// however, it is necessary because browsers will only open a payment
-		// sheet in direct response to user input (see
-		// https://developer.mozilla.org/en-US/docs/Web/API/PaymentRequest/show),
-		// so we can't delay the opening of the payment sheet on waiting for
-		// the response to this API call.
-		wpcom.undocumented().paygateConfiguration(
-			{
-				request_type: 'new_purchase',
-				country: selectedCountryCode,
-				// The endpoint also accepts an optional credit card brand in
-				// cases where the processor country might depend on the card
-				// brand used, but unfortunately we can't send that because we
-				// don't know it yet (it's a chicken and egg problem; the user
-				// only selects a credit card once the Apple Pay payment sheet
-				// is open, but we need to send Apple Pay the processor country
-				// in order to open the payment sheet). Fortunately, in most
-				// cases this won't have a practical effect.
-			},
-			function( configError, configuration ) {
-				let processorCountry = 'US';
-				if ( ! configError && configuration.processor === 'stripe_ie' ) {
-					processorCountry = 'IE';
-				}
-				this.setState( { processorCountry } );
-			}.bind( this )
-		);
-	};
+function getPostalCodeStringFromPostalCode( postalCode ) {
+	return typeof postalCode === 'undefined' || postalCode === null ? '' : postalCode;
+}
 
-	/**
-	 * @param {object} event  Event object.
-	 */
-	updateSelectedPostalCode = event => {
-		const { name: key, value } = event.target;
+function usePaymentRequestOptions( {
+	translate,
+	sub_total_integer,
+	total_tax_integer,
+	total_cost_integer,
+	countryCode,
+	currency,
+	shouldDisplayItems,
+} ) {
+	const displayItems = useMemo(
+		() => getDisplayItems( { translate, sub_total_integer, total_tax_integer } ),
+		[ translate, sub_total_integer, total_tax_integer ]
+	);
+	const paymentRequestOptions = useMemo(
+		() => ( {
+			country: countryCode || 'US',
```

Comment:
I think this may be sending the wrong country.  If I'm reading the code right, it looks like it's passing in the billing country, but based on https://stripe.com/docs/stripe-js/reference#stripe-payment-request it looks like it should be the Stripe account country.

In the old version, we had code that contacted `wpcom.undocumented().paygateConfiguration()` to get the best guess of which Automattic Stripe account (and therefore country) the payment would go through.  It's not clear to me if the same code would work here or not (and I'm also not sure offhand how `stripe.js` payments work with different Stripe accounts, compared to Paygate ones?).  Nor am I sure how important it is to send the correct country here... although I remember hearing somewhere that it can potentially cause Apple Pay's bypass of Strong Customer Authentication to fail.

Bottom line, though - it probably shouldn't be the billing country.

Reply:
I remember when I was first refactoring this section that I wasn't sure if it should be the billing or the processor country, but for some reason I decided it must be the billing country. It seems strange that it would want the processor country, since the Stripe object should already have that in its state. That said, 

> The two-letter country code of your Stripe account (e.g., 'US').

seems pretty clear that you're right. Weird. Ok, I can change it.

Reply:
Done.